PayloadPkg/FirmwareUpdate/FirmwareUpdate.c

```diff
@@ -1330,14 +1382,22 @@ InitFirmwareUpdate (
     return Status;
   }
 
+  // Get the number of payloads
+  CapHdr = (EFI_FW_MGMT_CAP_HEADER *)((UINTN)FwUpdHdr + FwUpdHdr->ImageOffset);
+  PayloadItemCount = CapHdr->PayloadItemCount;
+  if (PayloadItemCount > MAX_FW_COMPONENTS) {
+    DEBUG((DEBUG_ERROR, "# of payloads(%d) in capsule exceeds max(%d). The excess payloads will be skipped\n",
+          PayloadItemCount, MAX_FW_COMPONENTS));
+    PayloadItemCount = MAX_FW_COMPONENTS;
+  }
+
   //
   // Loop through the components to perform update
   //
-  for (Count = 0; Count < MAX_FW_COMPONENTS; Count ++) {
+  for (Count = 0; Count < PayloadItemCount; Count ++) {
```

Comment:
Is that required?  An unused FwUpdCompStatus[Count].UpdatePending == 0.

Assume MAX_FW_COMPONENTS is 10000, it will save certain cpu time, but for single-digit MAX_FW_COMPONENT, the change in line 1385 ~ 1393 adds more code complexity than its benefit.

Reply:
My idea is to process only the number of PayloadItemCount payloads.
Consider the situation when there is only one BIOS payload (means PayloadItemCount < MAX_FW_COMPONENTS), then we may need to check the UpdatePending status of the next FwUpdCompStatus[Count] to know if the BIOS payload is the final one to process.

Reply:
it is not neat. you can think twice about my comment/example left above.